pandas/tests/test_window.py

```diff
@@ -1228,242 +1177,169 @@ def roll_mean(x, window, min_periods=None, freq=None, center=False,
+                    20)[9:].reset_index(drop=True)
 
-            tm.assert_numpy_array_equal(result, expected)
+            tm.assert_series_equal(result, expected)
 
         if test_stable:
-            result = get_result(self.arr + 1e9, window)
+            result = get_result(self.series + 1e9, window)
             tm.assert_almost_equal(result[-1],
                                    static_comp(self.arr[-50:] + 1e9))
 
         # Test window larger than array, #7297
         if test_window:
             if has_min_periods:
-                for minp in (0, len(self.arr) - 1, len(self.arr)):
-                    result = get_result(self.arr, len(self.arr) + 1,
+                for minp in (0, len(self.series) - 1, len(self.series)):
+                    result = get_result(self.series, len(self.series) + 1,
                                         min_periods=minp)
-                    expected = get_result(self.arr, len(self.arr),
+                    expected = get_result(self.series, len(self.series),
                                           min_periods=minp)
-                    nan_mask = np.isnan(result)
-                    tm.assert_numpy_array_equal(nan_mask, np.isnan(expected))
+                    nan_mask = isna(result)
+                    tm.assert_series_equal(nan_mask, isna(expected))
 
                     nan_mask = ~nan_mask
                     tm.assert_almost_equal(result[nan_mask],
                                            expected[nan_mask])
             else:
-                result = get_result(self.arr, len(self.arr) + 1)
-                expected = get_result(self.arr, len(self.arr))
-                nan_mask = np.isnan(result)
-                tm.assert_numpy_array_equal(nan_mask, np.isnan(expected))
+                result = get_result(self.series, len(self.series) + 1)
+                expected = get_result(self.series, len(self.series))
+                nan_mask = isna(result)
+                tm.assert_series_equal(nan_mask, isna(expected))
 
                 nan_mask = ~nan_mask
                 tm.assert_almost_equal(result[nan_mask], expected[nan_mask])
 
-    def _check_structures(self, f, static_comp, name=None,
-                          has_min_periods=True, has_time_rule=True,
-                          has_center=True, fill_value=None, **kwargs):
-        def get_result(obj, window, min_periods=None, center=False):
+    def _check_moment_func(self, static_comp, name, has_min_periods=True,
+                           has_center=True, has_time_rule=True,
+                           fill_value=None, zero_min_periods_equal=True,
+                           **kwargs):
 
-            # check via the API calls if name is provided
-            if name is not None:
-                r = obj.rolling(window=window, min_periods=min_periods,
-                                center=center)
-                return getattr(r, name)(**kwargs)
+        self._check_series(name, static_comp=static_comp,
```

Comment:
I am not sure of the point of adding ``_check_series``, it *seems* that you are already checking the series & frame results (line 1344), yes?

Reply:
if there IS a difference, IOW you are checking something different, then certainly don't make a new function, just in-line it here.